MetalK8s UI: Fix wrong status in global health component

ChengYanJin opened the pull request:
**Component**: UI, dashboard

**Context**: 
In order to retrieve history alerts from LOKI Endpoint, we have to execute several queries in sequence, which causes the performance issue. 

So we switch to query Prometheus Endpoint to display the status of the Global Health Component. 

**Summary**:
Format the storage capacity in the PV list return by K8s Endpoint, to make sure the size has the correct unit.

We don't have data for the query we use to retrieve the number of ClusterAtRisk and ClusterDegraded, which cause the unavailable segment.

**Acceptance criteria**: 


---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #ISSUE_NUMBER

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

ChengYanJin committed:
ui/alerts: Sort the history alerts

To solve the bug that ClusterDegraded overrides the status the cluster
status which is set by ClusterAtRisk

Refs: ARTESCA-2206

ChengYanJin committed:
ui/loki: Do multiple query in a sequence to retrieve all the metrics

Given the limit number of metric can be contained in one query,
we may need to do multiple queries in a sequence.

Note that:
There may be performance issue

ChengYanJin committed:
ui: Remove the AlertHistoryProvider

Since we decided to query promethus to get the number of ClusterDegraded
and ClusterAtDisk which will be used by Global Health Component, then
there is no need to fetch the history alert for the moment, hence
removed the AlertHistoryProvider for the moment.

ChengYanJin committed:
ui/dashboard: Query prometheus to get the segments for GHC

ChengYanJin committed:
ui/tests: Fix the test for alertHistoryProvider

ChengYanJin committed:
ui/alerts: Remove the parents label

bert-e commented:
# Hello chengyanjin,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @alexandre-allard-scality

* @thomasdanan

* @gdemonet

* @chengyanjin

* @nicolast

* @jbwatenbergscality

* @teddyandrieux

* @lucieleonard

JBWatenbergScality commented on ui/src/components/DashboardGlobalHealth.js:
```diff
@@ -19,7 +19,9 @@ import { useStartingTimeStamp } from '../containers/StartTimeProvider';
 import CircleStatus, { StatusIcon } from './CircleStatus';
 import GlobalHealthBarComponent from '@scality/core-ui/dist/components/globalhealthbar/GlobalHealthBar.component';
-import { useHistoryAlerts } from '../containers/AlertHistoryProvider';
+import { getClusterAlertSegement } from '../services/platformlibrary/metrics';
```
Small typos and I would rename it as follows
```suggestion
import { getClusterAlertSegmentQuery } from '../services/platformlibrary/metrics';
```

JBWatenbergScality commented on ui/src/services/loki/api.js:
```diff
@@ -40,33 +41,85 @@ export function getLast7DaysAlerts(): Promise<Alert[]> {
 
   return lokiApiClient
     .get(
       //We set limit to 1000 because the default number of lines retrievable is 100 which is
```
Upgrade this comment, `4999 because it is the max retrievable in one call`

ChengYanJin commented:
/approve

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/9885ba02e0ad65e5d5ce19e4e962ff06676b37e9) did not succeed in branch bugfix/fix-wrong-status-in-global-health-component.




*The following options are set:* **approve**

ChengYanJin committed:
ui/volumes: Format the storage capacity for PV list

In some cases, we may not get the correct size unit in the PV list from
K8s Endpoint, so we format it here.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @alexandre-allard-scality

* @thomasdanan

* @gdemonet

* @chengyanjin

* @nicolast

* @jbwatenbergscality

* @teddyandrieux

* @lucieleonard










*The following options are set:* **approve**

JBWatenbergScality commented on ui/src/ducks/app/volumes.js:
```diff
@@ -250,6 +251,13 @@ export function* fetchPersistentVolumes(): Generator<
+      result.body.items.forEach((item) => {
+        bytesToSize(allSizeUnitsToBytes(item.spec.capacity.storage));
+      });
     yield put(setPersistentVolumesAction(result.body.items ?? []));
```
```suggestion
    const pvs = result?.body?.items?.map((item) => {
        item.spec.capacity.storage = bytesToSize(allSizeUnitsToBytes(item.spec.capacity.storage));
return item;
      });
    yield put(setPersistentVolumesAction(pvs ?? []));
```

ChengYanJin committed:
ui/dashboard: Fix the global health bug

Bug:
When we don't have any cluster alert, we considered as healthy segment.

Fix:
We combine with Watchdog alert to identify the unavailable time.

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/af4abc95b523cda109511e3a2e389c8c6a2f0a53) did not succeed in branch bugfix/fix-wrong-status-in-global-health-component.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.11`



The following branches will **NOT** be impacted:

* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.11`



The following branches have **NOT** changed:

* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



Please check the status of the associated issue None.

Goodbye chengyanjin.
